### PR TITLE
DC-458 Update Latency messages to include new values

### PR DESCRIPTION
**Pull request by FrancescoBuc86**
- Added TraceParentHeader Class and TraceParentHeaderGenerator
- Updated necessary fields
- Tests

### CODE REVIEW

File: plugin-common/src/main/java/org/opennms/plugins/cloud/grpc/TraceParentHeader.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.plugins.cloud.grpc;
+
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
```
Comment (patrick-schweizer) [inline plugin-common/src/main/java/org/opennms/plugins/cloud/grpc/TraceParentHeader.java:35]: I think we can replace the annotations with
`@Data` and `@Builder` and make all fields final:
We have then an immutable object with hashCode(), equals() toString() methods

File: plugin-common/src/test/java/org/opennms/plugins/cloud/testserver/GrpcTestServerInterceptor.java [hunk 2]
```diff
@@ -59,9 +65,13 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, Re
...
         // the following calls don't need a token, all others do:
-        return ! ("org.opennms.dataplatform.access.Authenticate/AuthenticateKey".equals(methodName)
+        return !("org.opennms.dataplatform.access.Authenticate/AuthenticateKey".equals(methodName)
                 || "org.opennms.dataplatform.access.Authenticate/GetServices".equals(methodName)
                 || "org.opennms.dataplatform.access.Authenticate/GetAccessToken".equals(methodName)
                 || "org.opennms.dataplatform.access.Authenticate/RenewCertificate".equals(methodName));
     }
+
+    private boolean isTraceParentHeaderNeeded(final String methodName) {
+        return StringUtils.equals(methodName, "org.opennms.tsaas.Timeseries/Store");
```
Comment (patrick-schweizer) [inline plugin-common/src/test/java/org/opennms/plugins/cloud/testserver/GrpcTestServerInterceptor.java:75]: I am not sure but I assume we want to have the trace for all GRPC calls, no?
I would assume we make that a general feature: Add the trace whenever we make a GRPC call?

File: plugin-common/src/main/java/org/opennms/plugins/cloud/grpc/TraceParentHeader.java (hunk 1 above)
Comment (FrancescoBuc86) [reply plugin-common/src/main/java/org/opennms/plugins/cloud/grpc/TraceParentHeader.java:35]: Yep ok

File: plugin-common/src/test/java/org/opennms/plugins/cloud/testserver/GrpcTestServerInterceptor.java (hunk 2 above)
Comment (FrancescoBuc86) [reply plugin-common/src/test/java/org/opennms/plugins/cloud/testserver/GrpcTestServerInterceptor.java:75]: Ok lets assume that

**Review by patrick-schweizer — DISMISSED** [review]

File: pom.xml [hunk 3]
```diff
@@ -36,7 +36,7 @@
         <awaitilityVersion>4.0.3</awaitilityVersion>
         <hamcrest.version>1.3</hamcrest.version>
         <tsaasClientVersion>1.0.0</tsaasClientVersion><!-- tsaas via cloud gateway -->
-        <telemetryGatewayVersion>1.0.0-feature-DC-391-6</telemetryGatewayVersion>
+        <telemetryGatewayVersion>1.0.0-feature-DC-391-23</telemetryGatewayVersion>
```
Comment (stumac) [inline pom.xml:39]: can update this to 1.2.2 if you'd prefer.

https://cloudsmith.io/~opennms/repos/maven/packages/detail/maven/telemetry-gateway-grpc/1.2.2/a=noarch;xg=org.opennms.tsaas.telemetry/

Comment (patrick-schweizer) [reply pom.xml:39]: :+1: that would look better

Comment (FrancescoBuc86) [reply pom.xml:39]: no problem

**Review by patrick-schweizer — APPROVED** [review]

**Pull request merged**